arangod/VocBase/Methods/Collections.cpp

```diff
@@ -180,30 +176,35 @@ Result validateCreationInfo(CollectionCreationInfo const& info,
     }
   }
 
-  return {TRI_ERROR_NO_ERROR};
+  return {};
 }
 
+// validate collection parameters. if the validation fails, it will audit-log
+// the failure.
 Result validateAllCollectionsInfo(
     TRI_vocbase_t const& vocbase,
     std::vector<CollectionCreationInfo> const& infos, bool allowSystem,
     bool allowEnterpriseCollectionsOnSingleServer,
     bool enforceReplicationFactor) {
+  Result res;
+
   for (auto const& info : infos) {
     // If the PlanId is not set, we either are on a single server, or this is
     // a local collection in a cluster; which means, it is neither a user-facing
     // collection (as seen on a Coordinator), nor a shard (on a DBServer).
 
     // validate the information of the collection to be created
-    Result res = validateCreationInfo(
+    res = validateCreationInfo(
```

Comment:
Actually it wasn't meant to be a performance optimization, because the runtime of this function does not matter much.
I intended to make the code more readable. But I agree, according to compiler explorer, the new variant's assembly code is disastrous.

Reply:
Changed to a better variant now.